File: src/main/java/burst/pool/storage/config/PropertyService.java
package burst.pool.storage.config;

import burst.kit.entity.BurstAddress;

public interface PropertyService {
    boolean getBoolean(Prop<Boolean> prop);
    int getInt(Prop<Integer> prop);
    long getLong(Prop<Long> prop);
    float getFloat(Prop<Float> prop);
    String getString(Prop<String> prop);
    String[] getStringList(Prop<String> prop);
    BurstAddress getBurstAddress(Prop<BurstAddress> prop);
}

File: src/main/kotlin/burst/pool/storage/config/PropertyService.kt
package burst.pool.storage.config

import burst.kit.entity.BurstAddress

interface PropertyService {
    /**
     * Get the value of the specified property, or the default value if it was not set
     * @param prop The property to get the value of
     * @return The value of [prop] or its default value if it is not set
     */
    fun <T> get(prop: Prop<T>): T
}
